tabular/src/autogluon/tabular/task/tabular_prediction/predictor_v2.py

```diff
@@ -0,0 +1,733 @@
+import copy
+import logging
+import math
+
+import numpy as np
+
+from autogluon.core.task.base import compile_scheduler_options
+from autogluon.core.task.base.base_task import schedulers
+from autogluon.core.utils import verbosity2loglevel
+from autogluon.core.utils.loaders import load_pkl
+from autogluon.core.utils.savers import save_pkl
+from autogluon.core.utils.utils import setup_outputdir, setup_compute, setup_trial_limits, default_holdout_frac
+
+from .dataset import TabularDataset
+from .hyperparameter_configs import get_hyperparameter_config
+from .predictor import TabularPredictor
+from .presets_configs import set_presets, unpack
+from ...features import AutoMLPipelineFeatureGenerator
+from ...learner import AbstractLearner, DefaultLearner
+from ...metrics import get_metric
+from ...trainer import AbstractTrainer, AutoTrainer
+
+
+logger = logging.getLogger()  # return root logger
+
+
+class TabularPredictorV2(TabularPredictor):
+    """
+    AutoGluon Task for predicting values in column of tabular dataset (classification or regression)
+    """
+    Dataset = TabularDataset
+    predictor_file_name = 'predictor.pkl'
+
+    def __init__(
```

Comment:
What's the logic to determine what goes here vs in `fit()`?
I'd imagine stuff that affects the models' quality (and could be viewed as a form of "hyperparameter") would go in `fit()`, but it seems some arguments like this are here instead, including: `label_count_threshold`, `feature_generator`, `feature_metadata`

Reply:
My thought process (happy to discuss) is currently:

In general, arguments which could break model fitting in repeated fit calls if changed go in the init, such as feature_generator, feature_metadata, and label_count_threshold (threshold determines which classes are used and how the data is split).

As they are optional arguments, I think we can also have a `.process_features` or `.fit_feature_generator` method to fit the generator prior to calling `.fit`.

1. label_count_threshold will likely live in init kwargs as it is (somewhat) tabular/stacking specific.
2. train_data could potentially live in init, but for now it lives in fit.
3. Once we support multiple feature generators, we could consider having feature_generator and feature_metadata live in fit. The logic to ensure that the stacker can work properly in these scenarios is complex though, particularly if we allow different train_data features in separate fit calls.

Reply:
Support for continuing training a model with additional epochs won't be available in v0.1 since it is very hard to universally apply and it is likely quite niche. I do agree than a way to overwrite an existing model would be useful (currently it will just rename the new model so it doesn't overwrite). However, that is also very niche and probably not necessary for v0.1.

2nd fit() will break if label_count_threshold changes because this impacts how many classes are actually outputted and the training data used. This will conflict with OOF preds and various other assumptions. Metadata and generator are not conflicting however, and can be changed as we discussed prior.